src/client/app/containers/app.ts

```diff
@@ -175,10 +175,26 @@ export class AppComponent {
                     this._store.dispatch(new GitHub.UpdateGistAction(this.snippet));
                 }
                 else if (isPublic) {
-                    this._store.dispatch(new GitHub.SharePublicGistAction(this.snippet));
+                    let sharePublicGist = async() => {
+                        let result = await this._effects.alert(Strings.sharePublicSnippetConfirm, `${Strings.share} ${this.snippet.name}`, `${Strings.share}`, `${Strings.cancelButtonLabel}`);
+                        return result;
```

Comment:
I think you can just return the result promise, no need to await for it here as we do not do anything with the return value in this function.